fortex/health/processors/scispacy_processor.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2022 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+SciSpacy Processor
+"""
+from typing import Dict, Set
+import importlib
+
+import spacy
+from forte.common import Resources
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.processors.base import PackProcessor
+
+# pylint: disable=unused-import
+from scispacy.abbreviation import AbbreviationDetector
+from scispacy.hyponym_detector import HyponymDetector
+
+from ftx.medical.clinical_ontology import Hyponym, Abbreviation, Phrase
+
+nlp = spacy.load("en_core_sci_sm")
+nlp.add_pipe("hyponym_detector", last=True, config={"extended": False})
+
+__all__ = [
+    "ScispaCyProcessor",
+]
+
+
+class ScispaCyProcessor(PackProcessor):
+    r"""
+    Implementation of this ScispaCyProcessor has been based on ScispaCy
```

Comment:
Suggest turning this into a new bug/new feature as this will be more complicated to implement and need more considerations other than the original bug/requirements.